inc/sainexthopgroup.h

```diff
@@ -128,6 +181,27 @@ typedef enum _sai_next_hop_group_member_attr_t
     SAI_NEXT_HOP_GROUP_MEMBER_ATTR_WEIGHT,
 
     /**
+     * @brief Preferred role in the protection group
+     *
+     * Should only be used if the type of owning group is SAI_NEXT_HOP_GROUP_TYPE_PROTECTION
```

Comment:
The problem with what you suggested is that you're trying to enforce a condition on an attribute on Next Hop Group Member based on the value of attribute of Next Hop Group.  Is there any way to express this with the @validonly syntax, or should I just add a comment saying something in the lines of "The field is only considered if the parent next hop group is PROTECTION.  Otherwise it is ignored"?